src/InstrumentationEngine.Lib/LoggerService.cpp

```diff
@@ -358,6 +358,7 @@ HRESULT CLoggerService::RecalculateLoggingFlags()
     }
 
     m_effectiveFlags = effectiveFlags;
+    CLogging::SetCachedLoggingFlag(effectiveFlags);
```

Comment:
>CLogging::SetCachedLoggingFlag(effectiveFlags); [](start = 4, length = 47)

This looks like a strange circular dependency. I didn't know that the logger service had access to CLogging. Seems like it shouldn't.

I was expecting that you add a way to register a callback on the CLoggerService for when the flags changed.

Reply:
@jander-msft what do you think of this? Is it good enough? You wrote this code initially.

Reply:
I was expecting callback registration as well. Though, this is probably good enough for now.

I think it would be nice to allow for IMs to do callback registration for logging flag changes as well so that they don't have to call IProfilerManagerLogging::GetLoggingFlags for each log operation, but that is probably a separate issue we'll want to address.

Reply:
I can implement the callback registration for this.